overlord/devicestate/devicestate_remodel_test.go

```diff
@@ -1975,3 +1978,549 @@ func (s *deviceMgrRemodelSuite) TestRemodelUC20LabelConflictsError(c *C) {
+			c.Check(filepath.Join(boot.InitramfsUbuntuBootDir, "device/model"),
+				testutil.FileContains, fmt.Sprintf("revision: %v\n", new.Revision()))
+		case 6:
+			c.Check(modeenv.Model, Equals, new.Model())
+			c.Check(modeenv.TryModel, Equals, "")
+			c.Check(filepath.Join(boot.InitramfsUbuntuBootDir, "device/model"),
+				testutil.FileContains, fmt.Sprintf("model: %s\n", new.Model()))
+			c.Check(filepath.Join(boot.InitramfsUbuntuBootDir, "device/model"),
+				testutil.FileContains, fmt.Sprintf("revision: %v\n", new.Revision()))
+		}
+		return nil
+	})
+	defer restore()
+
+	chg, err := devicestate.Remodel(s.state, new)
+	c.Assert(err, IsNil)
+
+	// since we cannot panic in random place in code that runs under
+	// taskrunner, we reset the task status and retry the change again, but
+	// we cannot do that once a change has become ready, thus inject a task
+	// that will request a reboot and keep retrying, thus stopping execution
+	// and keeping the change in a not ready state
+	fakeRebootCalls := 0
+	fakeRebootCallsReady := false
+	s.o.TaskRunner().AddHandler("fake-reboot-and-stall", func(task *state.Task, _ *tomb.Tomb) error {
+		fakeRebootCalls++
+		if fakeRebootCalls == 1 {
+			st := task.State()
+			st.Lock()
+			defer st.Unlock()
+			// not strictly needed, but underlines there's a reboot
+			// happening
+			st.RequestRestart(state.RestartSystemNow)
+		}
+		if fakeRebootCallsReady {
+			return nil
+		}
+		// we're not ready, so that the change does not complete yet
+		return &state.Retry{}
+	}, nil)
+	fakeRebootTask := s.state.NewTask("fake-reboot-and-stall", "fake reboot and stalling injected by tests")
+	chg.AddTask(fakeRebootTask)
+	var setModelTask *state.Task
+	for _, tsk := range chg.Tasks() {
+		if tsk.Kind() == "set-model" {
+			c.Fatalf("set-model present too early")
+		}
+		// make fake-reboot run after all tasks
+		if tsk.Kind() != "fake-reboot-and-stall" {
+			fakeRebootTask.WaitFor(tsk)
+		}
+	}
+	s.state.Unlock()
+
+	s.settle(c)
+
+	s.state.Lock()
+	// set model was injected by prepare-remodeling
+	for _, tsk := range chg.Tasks() {
+		if tsk.Kind() == "set-model" {
+			setModelTask = tsk
+			break
+		}
+	}
+	c.Check(chg.IsReady(), Equals, false)
+	c.Check(chg.Err(), IsNil)
+	// injected by fake restart
+	c.Check(s.restartRequests, DeepEquals, []state.RestartType{state.RestartSystemNow})
+	// 3 calls: promote tried system, old & new model, just the new model
+	c.Check(resealKeyCalls, Equals, 3)
+	// even if errors occur during reseal, set-model is done
+	c.Check(setModelTask.Status(), Equals, state.DoneStatus)
+
+	// reset the set-model state back to do, simulating a task restart after a reboot
+	setModelTask.SetStatus(state.DoStatus)
+
+	// the seeded systems has already been populated
+	var seededSystems []devicestate.SeededSystem
+	c.Assert(s.state.Get("seeded-systems", &seededSystems), IsNil)
+	c.Assert(seededSystems, HasLen, 2)
+	// time.Now() was not mocked, so we need to be smarted about checking
```

Comment:
```suggestion
	// time.Now() was not mocked, so we need to be smarter about checking
```

Reply:
That's fine, but curious if there is a reason we are not mocking timeNow in this test (FWTW devicestate has MockTimeNow)?